lib/Dialect/TritonNvidiaGPU/Transforms/RewriteTensorPointer.cpp

```diff
@@ -366,6 +366,9 @@ class TritonGPURewriteTensorPointerPass
       : computeCapability(computeCapability) {}
 
   static bool needRewrite(Operation *op, const DenseSet<Value> &valueToRemove) {
+    if (auto ifOp = dyn_cast<scf::IfOp>(op)) {
+      op = ifOp.thenYield().getOperation();
+    }
```

Comment:
what will happen if the IfOp also has an "else region"?

Reply:
ignore, only the types are needed to be considered here.

Reply:
This IfOp should be rewritten as long as the yield op of one of its then block and else block contains an operand that can be rewritten. And I suppose then yield and else yield depend on the same MakeTensorPtrOp, but it is not safe enough. I will revise latter, Thanks.

Reply:
done